packages/wrangler/src/__tests__/dev.test.tsx

```diff
@@ -1,31 +1,23 @@
-import { render } from "ink-testing-library";
+import * as fs from "node:fs";
 import patchConsole from "patch-console";
-import React from "react";
 import Dev from "../dev";
+import { mockAccountId, mockApiToken } from "./helpers/mock-account-id";
+import { setMockResponse, unsetAllMocks } from "./helpers/mock-cfetch";
 import { mockConsoleMethods } from "./helpers/mock-console";
+import { runInTempDir } from "./helpers/run-in-tmp";
 import { runWrangler } from "./helpers/run-wrangler";
 import writeWranglerToml from "./helpers/write-wrangler-toml";
-import type { DevProps } from "../dev";
 
 describe("Dev component", () => {
-  let restoreConsole: ReturnType<typeof patchConsole>;
-  beforeEach(() => (restoreConsole = patchConsole(() => {})));
-  afterEach(() => restoreConsole());
+  beforeEach(() => {});
+  mockAccountId();
+  mockApiToken();
+  runInTempDir();
   const std = mockConsoleMethods();
-
-  // This test needs to be rewritten because the error now throws asynchronously
-  // and the Ink framework does not yet have async testing support.
-  it.skip("should throw if format is service-worker and there is a public directory", () => {
-    const { lastFrame } = renderDev({
-      format: "service-worker",
-      accountId: "some-account-id",
-      public: "some/public/path",
-    });
-    expect(lastFrame()?.split("\n").slice(0, 2).join("\n"))
-      .toMatchInlineSnapshot(`
-      "Something went wrong:
-      Error: You cannot use the service worker format with a \`public\` directory."
-    `);
```

Comment:
Fair enough. This test is not being used, nor is it very helpful.
We do need to work on some way of automating our testing of `wrangler dev` in general though.